src/lib/firebase/client/firestore/users.ts

```diff
@@ -3,7 +3,15 @@ import { collection, getDocs, query } from 'firebase/firestore'
 import type { user } from '$lib/types/domain/user'
 
 /** Get data of the given user ids */
-export async function GetAllUsers() {
+export async function getUsers(ids?: string[]) {
 	const usersQuery = query(collection(db, 'users'))
-	return (await getDocs(usersQuery)).docs.map((user) => ({ id: user.id, ...user.data() } as user))
+	const users = (await getDocs(usersQuery)).docs.map(
+		(user) => ({ id: user.id, ...user.data() } as user)
+	)
+
+	// If no ids are given, return all users
+	if (!ids) return users
+
+	// Otherwise, filter the users
+	return users.filter((user) => ids.includes(user.id))
```

Comment:
Deze filtering wil je server side (met where), en je moet dan wrs de "batched in" util functie gebruiken omdat je niet een in query met meer dan 10 items mag doen